### PR TITLE
Delete config 'disable_bg_flush '

### LINKED ISSUE #4641: Remove useless `disable_bg_flush` configuration
## Enhancement
In the early version of TiFlash, we build a TxnMergeTree (Txn for Transaction) base on ClickHouse's MergeTree to adapt the transaction model for TiDB. However, DeltaTree is more suitable for TiFlash under HTAP scenario and we remove TxnMergeTree at last.
However, in the early version, we add the option `disable_bg_flush` to adapt the different behavior between TxnMergeTree and DeltaTree. Now the TxnMergeTree is removed, and we should cleanup the useless `disable_bg_flush` option from TiFlash.

### PR DESCRIPTION
**Pull request by hzh0425**
### What problem does this PR solve?

Issue Number: close #4641

Problem Summary: In the early version, we add the option disable_bg_flush to adapt the different behavior between TxnMergeTree and DeltaTree. Now the TxnMergeTree is removed, and we should cleanup the useless disable_bg_flush option from TiFlash.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Transaction/KVStore.cpp [hunk 1]
```diff
@@ -129,35 +129,32 @@ void KVStore::traverseRegions(std::function<void(RegionID, const RegionPtr &)> &
...
-        auto table_id = region.getMappedTableID();
-        auto storage = tmt.getStorages().get(table_id);
-        if (storage == nullptr)
-        {
-            LOG_WARNING(log,
-                        "tryFlushRegionCacheInStorage can not get table for region:" + region.toString()
-                            + " with table id: " + DB::toString(table_id) + ", ignored");
-            return;
-        }
+        LOG_WARNING(log,
```
[the remark this answers is not in the document]
Comment (ywqzzy) [reply dbms/src/Storages/Transaction/KVStore.cpp:136]: LOG_FMT_WARNING

File: dbms/src/Storages/Transaction/KVStore.cpp [hunk 2]
```diff
@@ -445,21 +442,13 @@ EngineStoreApplyRes KVStore::handleAdminRaftCmd(raft_cmdpb::AdminRequest && requ
...
-                    tmt.getRegionTable().tryFlushRegion(region, false);
-                }
-                catch (const Exception & e)
-                {
-                    tryLogCurrentException(__PRETTY_FUNCTION__);
-                }
+                tmt.getRegionTable().tryFlushRegion(region, false);
             }
-            else
+            catch (const Exception & e)
```
Comment (SeaRise) [inline dbms/src/Storages/Transaction/KVStore.cpp:449]: how about
`catch (...)`?

File: dbms/src/Storages/Transaction/KVStore.cpp [hunk 3]
```diff
@@ -129,35 +129,32 @@ void KVStore::traverseRegions(std::function<void(RegionID, const RegionPtr &)> &
 
 void KVStore::tryFlushRegionCacheInStorage(TMTContext & tmt, const Region & region, Poco::Logger * log)
 {
-    if (tmt.isBgFlushDisabled())
+    auto table_id = region.getMappedTableID();
+    auto storage = tmt.getStorages().get(table_id);
+    if (storage == nullptr)
```
Comment (SeaRise) [inline dbms/src/Storages/Transaction/KVStore.cpp:134]: how about `if (unlikey(storage == nullptr))`?

File: dbms/src/Storages/Transaction/BackgroundService.cpp [hunk 4]
```diff
@@ -36,65 +36,12 @@ BackgroundService::BackgroundService(TMTContext & tmt_)
         },
         false);
 
-    if (!tmt.isBgFlushDisabled())
-    {
-        table_flush_handle = background_pool.addTask([this] {
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Transaction/BackgroundService.cpp:41]: `region_mutex`/`regions_to_flush`/`table_flush_handle`/`region_handle` should also be removed
https://github.com/pingcap/tiflash/blob/e1db4677cef8edcb0a573655d659bad0bb451535/dbms/src/Storages/Transaction/BackgroundService.h#L48-L53

Comment (JaySon-Huang) [reply dbms/src/Storages/Transaction/BackgroundService.cpp:41]: Check whether `RegionTable::pickRegionToFlush` can be clean or not after we remove these logic

Comment (hzh0425) [reply dbms/src/Storages/Transaction/BackgroundService.cpp:41]: okay, thanks

File: dbms/src/Storages/Transaction/KVStore.cpp (hunk 2 above)
Comment (hzh0425) [reply dbms/src/Storages/Transaction/KVStore.cpp:449]: Hi, thanks for reivew. but I don't understand what you mean, can you explain it in detail? Thanks a lot.

File: dbms/src/Storages/Transaction/KVStore.cpp [hunk 5]
```diff
@@ -27,6 +27,7 @@
 #include <Storages/Transaction/RegionExecutionResult.h>
 #include <Storages/Transaction/RegionTable.h>
 #include <Storages/Transaction/TMTContext.h>
+#include "common/likely.h"
```
Comment (ywqzzy) [inline dbms/src/Storages/Transaction/KVStore.cpp:30]: maybe not necessary.
or change the format of include.

Comment (ywqzzy) [reply dbms/src/Storages/Transaction/KVStore.cpp:30]:
```suggestion
#include <common/likely.h>
```

Comment (hzh0425) [reply dbms/src/Storages/Transaction/KVStore.cpp:30]: oh yep, thanks a lot

**Review by JaySon-Huang — APPROVED** [review]
LGTM

File: dbms/src/Storages/Transaction/KVStore.cpp [hunk 6]
```diff
@@ -129,35 +130,33 @@ void KVStore::traverseRegions(std::function<void(RegionID, const RegionPtr &)> &
...
-        {
-            LOG_WARNING(log,
-                        "tryFlushRegionCacheInStorage can not get table for region:" + region.toString()
-                            + " with table id: " + DB::toString(table_id) + ", ignored");
-            return;
-        }
+        LOG_FMT_WARNING(log,
+                        "tryFlushRegionCacheInStorage can not get table for region {} with table id {}, ignored",
+                        region.toString(),
+                        DB::toString(table_id));
```
Comment (SeaRise) [inline dbms/src/Storages/Transaction/KVStore.cpp:140]:
```suggestion
                        table_id);
```

Comment (hzh0425) [reply dbms/src/Storages/Transaction/KVStore.cpp]: Okay, thanks a lot

**Comment by JaySon-Huang** [comment]
@ywqzzy @SeaRise PTAL

**Pull request merged**